charger/openevse.go

```diff
@@ -0,0 +1,446 @@
+package charger
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/charger/openevse"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// OpenEVSE charger implementation
+type OpenEVSE struct {
+	uri    string
+	phases int
+	api    *openevse.ClientWithResponses
+	log    *util.Logger
+	api.MeterEnergy
+}
+
+func init() {
+	registry.Add("openevse", NewOpenEVSEFromConfig)
+}
+
+// go:generate go run oapi-codegen -package openevse -old-config-style -generate "types,client" openevse/api.yaml > openevse/api.go
```

Comment:
Generator doesnt work

Reply:
I'll remove it since it requires `oapi-codegen` to be installed.
Should I add it to the Makefile?

Reply:
Please leave it in and add codegen to tools.go